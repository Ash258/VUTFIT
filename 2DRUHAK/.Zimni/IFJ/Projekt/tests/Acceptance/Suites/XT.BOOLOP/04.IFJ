' GCD of Two Numbers using Recursion

declare function modulo(a as integer, b as integer) as integer

function hcf(n1 as integer, n2 as integer) as integer
    if n2 <> 0 then
        dim vysledek as integer
        dim modulo_ as integer
        modulo_ = modulo(n1, n2)
        vysledek = hcf(n2, modulo_)
       return vysledek
    else
       return n1
    end if
end function

function modulo(delitel as integer, delenec as integer) as integer
    dim vysledek as integer
    vysledek = delitel - (delitel\delenec)*delenec
    return vysledek
end function

scope
    dim n1 as integer = 366
    dim n2 as integer = 60

    dim vysledek as integer
    vysledek = hcf(n1, n2)

    print !"G.C.D of"; n1; !" and"; n2 ; !" is"; vysledek;
end scope
