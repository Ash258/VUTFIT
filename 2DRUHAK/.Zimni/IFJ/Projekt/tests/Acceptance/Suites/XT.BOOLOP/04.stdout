G.C.D of 366 and 60 is 6